File: python3/2405.py
class Solution:
    def partitionString(self, s: str) -> int:
        c = 1
        f = 0
        for i in s:
            x = 1 << (ord(i) - 97)
            if f & x:
                f = 0
                c += 1
            f |= x
        return c

File: dart/2405.dart
class Solution {
    int partitionString(String s) {
        int n = s.length, c = 1, f = 0;
        String l = "abcdefghijklmnopqrstuvwxyz";
        for (int i = 0; i < n; i++) {
            int x = 1 << (l.indexOf(s[i]));
            if ((f & x) != 0) {
                f = 0;
                c++;
            }
            f |= x;
        }
        return c;
    }
}
